lib/src/rules/unnecessary_final.dart

```diff
@@ -10,9 +10,13 @@ import '../analyzer.dart';
 const _desc = "Don't use `final` for local variables.";
 
 const _details = r'''
-**DON'T** use `final` for local variables.
+Use `var`, not `final`, when declaring local variables.
 
-`var` is shorter, and `final` does not change the meaning of the code.
+Per Effective Dart, there are two styles in wide use. This rule
```

Comment:
Link to the Effective Dart rule?

https://dart.dev/guides/language/effective-dart/usage#do-follow-a-consistent-rule-for-var-and-final-on-local-variables

Not sure if/how linking works here though.

Reply:
Found an example in the `comment_references` lint

https://dart-lang.github.io/linter/lints/comment_references.html

looks like standard markdown, square brackets for the link text then rounded brackets for the link

```dart
const _details = r'''
**DO** reference only in scope identifiers in doc comments.

If you surround things like variable, method, or type names in square brackets,
then [`dart doc`](https://dart.dev/tools/dart-doc) will look
up the name and link to its docs.  For this all to work, ensure that all
identifiers in docs wrapped in brackets are in scope.
```

Reply:
Nice, added the link!

Reply:
@mit-mit this is good. It's what I had in mind, and it's not towards what I think all the documentation should have: a reference to effective Dart